### ISSUE TITLE
Status van gebouw of gebouweenheid / lifecycle

### ISSUE THREAD

**Issue by gezever** [Informatievlaanderen/OSLO-Discussion#332]
**gebouwenregister applicatieprofiel / vocabularium OSLO-Gebouw** 
 
**Omschrijving van het probleem:** 
Volgens het applicatieprofiel heeft een gebouw en een gebouweenheid exact één gebouw(eenheid)status. Indien een gebouw(eenheid) een unieke identifier over de volledige levenscyclus heeft,  betekent dit volgens mij dat er volgens dit model geen historische gegevens mogelijk zijn.  
Is dit de bedoeling?
**Omschrijving van een mogelijke oplossing:**

**Comment by bertvannuffelen** [comment]
Nee, zo moet je dat niet lezen. Er is slechts 1 status voor een toestand in de tijd. Een Gebouw is een subklasse van [Object](https://otl-test.data.vlaanderen.be/doc/applicatieprofiel/generiek-basis/#Object) ( in https://otl-test.data.vlaanderen.be/doc/applicatieprofiel/generiek-basis/) 

De identificator verwijst naar de huidige (laatst bekende) toestand van de werkelijkheid. De verschillende toestanden (die geregistreerd zijn) vormen _versies_ van hetzelfde object.

**Comment by GeertThijs** [comment]
Historiek wordt buiten de semantische standaard gehouden. Het wordt beschouwd als een aspect van implementatie. Om bij jouw vb te blijven: als we aan status een kardinaliteit van 1..* zouden geven zou immers de indruk gewekt worden dat een Gebouweenheid meerdere statussen tegelijk kan hebben, wat hier niet het geval is. Het verstoort de werkelijke semantiek. Wat we wel hebben voorzien is een UML-profiel voor historiek. dat laat toe om bij het uitwerken ve implementatiemodel stereotypes op de elementen te plakken, voor status zou dat bv kunnen zijn: opeenvolgende versies met hun geldigheidsperiode volgens geldigetijd. In https://github.com/Informatievlaanderen/OSLO-handleiding/blob/master/Implementatiemodellen/Historiek.pdf geef ik geheel toevallig hetzelfde voorbeeld als jij.

**Comment by gezever** [comment]
Mij lijkt dat enkel de versies een toestand op een bepaald moment hebben en het 'gebouw' op zich niet. Als iemand enkel de huidige toestand wil kennen, kan die toch een 'view' queryen.  Zou je de kardinaliteit dan niet 0..1 maken? verder vind ik het ook niet praktisch om steeds updates te moeten doen op iets dat eigenlijk geen property van een gebouw is, maar een eigenschap dat toegekend wordt, afgeleid wordt, berekend kan worden. Analoog aan leeftijd voor een persoon ga je die waarde afleiden van de huidige datum en de geboortedatum. Voor een gebouw kan je eender welke status toekennen, afhankelijk van activiteiten. bv. de toestand 'pas geschilderd' kan je afleiden van het feit dat een activiteit 'schilderen' de ene versie naar een andere versie heeft gebracht.

**Comment by GeertThijs** [comment]
> Mij lijkt dat enkel de versies een toestand op een bepaald moment hebben en het 'gebouw' op zich niet. Als iemand enkel de huidige toestand wil kennen, kan die toch een 'view' queryen. Zou je de kardinaliteit dan niet 0..1 maken? verder vind ik het ook niet praktisch om steeds updates te moeten doen op iets dat eigenlijk geen property van een gebouw is, maar een eigenschap dat toegekend wordt, afgeleid wordt, berekend kan worden. Analoog aan leeftijd voor een persoon ga je die waarde afleiden van de huidige datum en de geboortedatum. Voor een gebouw kan je eender welke status toekennen, afhankelijk van activiteiten. bv. de toestand 'pas geschilderd' kan je afleiden van het feit dat een activiteit 'schilderen' de ene versie naar een andere versie heeft gebracht.

- Een Versie van een Object volgt uit het feit dat zijn werkelijke toestand (GeldigeTijd) of die van zijn representatie in de databank verandert (Transactietijd). Als toestand zou slaan op de representatie (bv Ingevoerd, Bijgewerkt...) ben ik het met je eens, echter als de toestand slaat op het Object op het terrein (bv Inaanbouw, Afgebroken, InGebruik... ) dan klopt jouw redenering niet. In dat tweede geval is status (of toestand als je wil) wel degelijk een kenmerk van het Object.
- En inderdaad wordt historiek in een databank doorgaans dmv een view geïmplementeerd, uit een set van Versies wordt dan de actuele Versie gepikt. Maar is dat een reden om de kardinaliteit op 0..1 ipv 1 te zetten? Dat de kardinaliteit 1 is in het Gebouwenregister is een kwestie van semantiek: een Gebouw heeft altijd een status, maar bv niet noodzakelijk een adres. 
- Dat sommige kenmerken uit andere afgeleid kunnen woren klopt (aan te geven met /-teken in UML). Voor status zou dat kunnen als je bv ook Events die plaatsvinden op de Objecten zou opslaan. Typisch voor een event Sourcing architectuur, maar ook daar slaan de Events doorgaans enkel op databank gebeurtenissen. Het is niet omdat zijn record logisch op "verwijderd" wordt gezet dat een Gebouw afgebroken is op terrein (het is misschien gefusioneerd met het naastliggend Gebouw).
